src/tests/roc_fec/target_openfec/test_writer_reader.cpp

```diff
@@ -902,5 +902,122 @@ TEST(writer_reader, repaired_block_numbering) {
     }
 }
 
+TEST(writer_reader, resize_souce_block_increase_decrease) {
```

Comment:
There is also another concern with this test. If both writer and reader are broken, and the writer will correctly fill the SBL field but will not really change the block size, and the reader will just ignore the SBL field, this test will nevertheless happily succeed. It would be great if our test suite would also check that block sizes are *really* changing.

One way to do it is to add a test checking that the writer produces correct output when we're changing the block size. We set it to `<10, 5>` and it produces 10 source then 5 repair packets. We set it to `<15, 5>` and it produces 15 source then 5 repair packets.

In that way we would have 1) a test checking that writer produces an expected output 2) a test checking that writer + reader work as expected together (resize_souce_block_increase_decrease). This pair of tests would be enough to be sure that writer and reader work properly both individually and in pair.

Reply:
> One way to do it is to add a test checking that the writer produces correct output when we're changing the block size.

BTW the test I suggested should be added to test_encoder_decoder instead of test_writer_reader.

Reply:
> It would be great if our test suite would also check that block sizes are really changing.
> We set it to <10, 5> and it produces 10 source then 5 repair packets. We set it to <15, 5> and it produces 15 source then 5 repair packets.

Note, that test validates that the source block size has really changed:

```
        for (size_t i = 0; i < block_sizes[n]; ++i) {
            const packet::PacketPtr p = reader.read();

            CHECK(p);
            CHECK(p->fec());
            CHECK(p->fec()->source_block_length == block_sizes[n]);
        }
```

We just need to ensure that we have the same behavior in case of repair packets.

Reply:
> BTW the test I suggested should be added to test_encoder_decoder instead of test_writer_reader.

It is not clear why

Reply:
I've added a test `TEST(writer_reader, lossy_resize_source_block)` that somehow verifies the correctness of the block resizing. I think it is enough for the current PR.

Reply:
> Note, that the test validates that the source block size has really changed:

It check the SBL *field*, but not the *real* block size.

> It is not clear why.

My bad, I was wrong with that.

> I've added a test TEST(writer_reader, lossy_resize_source_block) that somehow verifies the correctness of the block resizing. I think it is enough for the current PR.

LGTM.